### PR TITLE
feat: ActivityIndicator component

### PR DESCRIPTION
**Pull request by jaulz**
### Motivation

This component serves as a drop-in replacement for the component that is delivered by React Native out of the box. It's design tries to imitate the indeterminate circular progress indicator from the Material guidelines:
https://material.io/design/components/progress-indicators.html#circular-progress-indicators 

### Test plan

Open examples and see examples of `ActivityIndicator`.

### CODE REVIEW

File: example/src/ActivityIndicatorExample.js [hunk 1]
```diff
@@ -0,0 +1,84 @@
+/* @flow */
+
+import * as React from 'react';
+import { View, StyleSheet } from 'react-native';
+import { Colors, FAB, withTheme, type Theme } from 'react-native-paper';
+import { ActivityIndicator } from '../../src';
```
Comment (Trancever) [inline example/src/ActivityIndicatorExample.js]: Can we import it from 'react-native-paper' instead?

File: src/components/ActivityIndicator.js [hunk 2]
```diff
@@ -0,0 +1,269 @@
+/* @flow */
+
+import * as React from 'react';
+import { Animated, Easing, Platform, StyleSheet, View } from 'react-native';
+import { withTheme } from '../core/theming';
+import type { Theme } from '../../types';
+
+type Props = {|
```
Comment (Trancever) [inline src/components/ActivityIndicator.js]: I think in future we should add `type` prop for Determinate and Indeterminate states, but it shouldn't be breaking change so we can keep that API as it is.

Comment (jaulz) [reply src/components/ActivityIndicator.js]: You are right but at the moment this component is supposed to be a drop in replacement for the native component. Anyway in the future we could enhance it and just make `type` as `indeterminate` as default.

File: example/src/ActivityIndicatorExample.js (hunk 1 above)
Comment (jaulz) [reply example/src/ActivityIndicatorExample.js]: Done

**Pull request merged**